### PR TITLE
hw-03

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer object : manufacturers) {
+            if (object.getId() == manufacturer.getId()) {
+                Storage.manufacturers.set(Math.toIntExact(object.getId() - 1L), manufacturer);
```
Comment (gigbat) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: it's up to you, but method set() have int as parameterm and when you have more digital than int you will get some erros. I would write like this: 
```suggestion
                Storage.manufacturers.remove(object.getId());
                Storage.manufacturers.add(manifacturer);

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                return manufacturer;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (int i = 0; i < manufacturers.size(); i++) {
```
Comment (gigbat) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: replace to long and write Storage.manufacturers.remove(Object)  just) and removed Math.toIntExact

File: src/main/java/core/basesyntax/model/Car.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.List;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
```
Comment (gigbat) [inline src/main/java/core/basesyntax/model/Car.java]: check for null

File: src/main/java/core/basesyntax/model/Car.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getModel() {
+        return model;
+    }
+
+    public void setModel(String model) {
```
Comment (gigbat) [inline src/main/java/core/basesyntax/model/Car.java:28]: same check for null

File: src/main/java/core/basesyntax/model/Car.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    public void setManufacturer(Manufacturer manufacturer) {
+        this.manufacturer = manufacturer;
+    }
+
+    public List<Driver> getDrivers() {
+        return drivers;
+    }
+
+    public void setDrivers(List<Driver> drivers) {
```
Comment (gigbat) [inline src/main/java/core/basesyntax/model/Car.java:44]: null)

File: src/main/java/core/basesyntax/model/Car.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    public void setModel(String model) {
+        this.model = model;
+    }
+
+    public Manufacturer getManufacturer() {
+        return manufacturer;
+    }
+
+    public void setManufacturer(Manufacturer manufacturer) {
```
Comment (gigbat) [inline src/main/java/core/basesyntax/model/Car.java:36]: same

**Review by vlykhvar — APPROVED** [review]

**Review by gigbat — CHANGES_REQUESTED** [review]
Good job 👍

**Review by gigbat — APPROVED** [review]

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer manufacturer : manufacturers) {
+            if (manufacturer.getId() == id) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: what is the correct way to compare two objects?

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer manufacturer : manufacturers) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: let's use stream, that will make code shorter

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer object : manufacturers) {
+            if (object.getId() == manufacturer.getId()) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: same with the comparions

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java (hunk 1 above)
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: use indexes and when you've found needed object with the same id call method set(index)

File: src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+                Storage.manufacturers.set(Math.toIntExact(object.getId() - 1L), manufacturer);
+                return manufacturer;
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java]: better to use method removeIf here

File: src/main/java/core/basesyntax/model/Car.java (hunk 3 above)
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/model/Car.java]: checklist

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**